hack/ci/run-default-application-e2e-test.sh

```diff
@@ -0,0 +1,104 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+### This script is used as a postsubmit job and updates the dev master
+### cluster after every commit to main.
+
+set -euo pipefail
+
+cd $(dirname $0)/../..
+source hack/lib.sh
+
+TEST_NAME="Pre-warm Go build cache"
+echodate "Attempting to pre-warm Go build cache"
+
+# Directly use environment variables set earlier in the script or passed in.
+APPLICATION_INSTALLATION_NAME="${APPLICATION_INSTALLATION_NAME:-}"
+APPLICATION_NAME="${APPLICATION_NAME:-}"
+APPLICATION_NAMESPACE="${APPLICATION_NAMESPACE:-}"
+APP_LABEL_KEY="${APP_LABEL_KEY:-}"
+NAMES="${NAMES:-}"
+
+# Ensure all environment variables are set
+if [[ -z "$APPLICATION_INSTALLATION_NAME" || -z "$APPLICATION_NAME" || -z "$APPLICATION_NAMESPACE" || -z "$APP_LABEL_KEY" || -z "$NAMES" ]]; then
+  echo "One or more required environment variables are missing."
+  exit 1
+fi
+
+echo "Running test with application name '$APPLICATION_NAME' and version '$APPLICATION_NAME'"
+
+beforeGocache=$(nowms)
+make download-gocache
+pushElapsed gocache_download_duration_milliseconds $beforeGocache
+
+export KIND_CLUSTER_NAME="${SEED_NAME:-kubermatic}"
+export KUBERMATIC_YAML=hack/ci/testdata/kubermatic.yaml
+source hack/ci/setup-kind-cluster.sh
+
+# gather the logs of all things in the cluster control plane and in the Kubermatic namespace
+protokol --kubeconfig "$KUBECONFIG" --flat --output "$ARTIFACTS/logs/cluster-control-plane" --namespace 'cluster-*' > /dev/null 2>&1 &
+protokol --kubeconfig "$KUBECONFIG" --flat --output "$ARTIFACTS/logs/kubermatic" --namespace kubermatic > /dev/null 2>&1 &
+
+kubectl apply -f pkg/crd/k8c.io
+export INSTALLER_FLAGS="--deploy-default-app-catalog"
+source hack/ci/setup-kubermatic-in-kind.sh
+
+export GIT_HEAD_HASH="$(git rev-parse HEAD | tr -d '\n')"
+
+if [ -z "${E2E_SSH_PUBKEY:-}" ]; then
+  echodate "Getting default SSH pubkey for machines from Vault"
+  retry 5 vault_ci_login
+  E2E_SSH_PUBKEY="$(mktemp)"
+  vault kv get -field=pubkey dev/e2e-machine-controller-ssh-key > "${E2E_SSH_PUBKEY}"
+else
+  E2E_SSH_PUBKEY_CONTENT="${E2E_SSH_PUBKEY}"
+  E2E_SSH_PUBKEY="$(mktemp)"
+  echo "${E2E_SSH_PUBKEY_CONTENT}" > "${E2E_SSH_PUBKEY}"
+fi
+
+echodate "SSH public key will be $(head -c 25 ${E2E_SSH_PUBKEY})...$(tail -c 25 ${E2E_SSH_PUBKEY})"
+
+echodate "Running $APPLICATION_NAME tests..."
+
+pathToFile="pkg/ee/default-application-catalog/applicationdefinitions/$APPLICATION_NAME-app.yaml"
+echodate "File path: $pathToFile"
+
+if [[ -s "$pathToFile" ]]; then
+  versions=$(yq eval '.spec.versions[0].version' "$pathToFile")
```

Comment:
This will only give you the first item in the array aka the first version.

Which we can also see the test only runs against the first version, not all of them

For example for Argo (should be 3 versions)
![grafik](https://github.com/user-attachments/assets/052d5b4d-dd83-4438-a7fe-cd3068112a62)

I think this should work
```suggestion
  versions=$(yq eval '.spec.versions[].version' "$pathToFile")
```